### PR TITLE
added searchCarsController

### PR DESCRIPTION
**Pull request by VSemenovykh**

### CODE REVIEW

File: src/main/java/ru/ncedu/controller/RESTfullController.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import ru.ncedu.service.MotorService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@CrossOrigin(origins = {"http://localhost:4200"})
+@RequestMapping("/api")
+public class RESTfullController {
```
Comment (AndrewSmrnv) [inline src/main/java/ru/ncedu/controller/RESTfullController.java:21]: Let's change controller's name to something meaningful

File: src/main/java/ru/ncedu/controller/SearchCarsController.java [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+import ru.ncedu.service.AutoService;
+import ru.ncedu.service.BrandService;
+import ru.ncedu.service.MotorService;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@CrossOrigin(origins = {"http://localhost:4200"})
+@RequestMapping({"/api/search"})
```
Comment (AndrewSmrnv) [inline src/main/java/ru/ncedu/controller/SearchCarsController.java:19]: Curve brackets can be removed

File: src/main/java/ru/ncedu/controller/SearchCarsController.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+import ru.ncedu.model.AutoJoin;
+import ru.ncedu.service.AutoService;
+import ru.ncedu.service.BrandService;
+import ru.ncedu.service.MotorService;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@CrossOrigin(origins = {"http://localhost:4200"})
```
Comment (AndrewSmrnv) [inline src/main/java/ru/ncedu/controller/SearchCarsController.java:18]: Do we need this annotation?

File: src/main/java/ru/ncedu/controller/RESTfullController.java (hunk 1 above)
Comment (AndrewSmrnv) [inline src/main/java/ru/ncedu/controller/RESTfullController.java:21]: Please extract all business logic from the controller to corresponding services.

File: src/main/java/ru/ncedu/controller/SearchCarsController.java [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+@RequestMapping({"/api/search"})
+@Slf4j
+public class SearchCarsController {
```
Comment (AndrewSmrnv) [inline src/main/java/ru/ncedu/controller/SearchCarsController.java:21]: Please extract all business logic from the controller to corresponding services.

File: src/main/java/ru/ncedu/controller/SearchCarsController.java (hunk 3 above)
Comment (VSemenovykh) [reply src/main/java/ru/ncedu/controller/SearchCarsController.java]: yeas, do need

File: src/main/java/ru/ncedu/controller/SearchCarsController.java (hunk 4 above)
Comment (VSemenovykh) [reply src/main/java/ru/ncedu/controller/SearchCarsController.java]: done!

File: src/main/java/ru/ncedu/controller/RESTfullController.java (hunk 1 above)
Comment (VSemenovykh) [reply src/main/java/ru/ncedu/controller/RESTfullController.java]: done!

File: src/main/java/ru/ncedu/controller/SearchCarsController.java (hunk 2 above)
Comment (VSemenovykh) [reply src/main/java/ru/ncedu/controller/SearchCarsController.java]: done!

File: src/main/java/ru/ncedu/controller/RESTfullController.java (hunk 1 above)
Comment (VSemenovykh) [reply src/main/java/ru/ncedu/controller/RESTfullController.java]: done, i did split the RESTfullController on CreateAutoController, UpdateAutoController, DeleteAutoController and GetListAutoController

**Pull request merged**